proposals/handler.go

```diff
@@ -432,13 +433,17 @@ func (h *Handler) setProposalBeacon(p *types.Proposal) error {
 	return nil
 }
 
-func (h *Handler) processBallot(ctx context.Context, logger log.Log, b *types.Ballot) (*wire.MalfeasanceProof, error) {
+func (h *Handler) processBallot(
+	ctx context.Context,
+	logger *zap.Logger,
```

Comment:
What's the reason for passing along the logger? It can also be accessed via `h.logger` I believe?

Reply:
I think the reason is it has some fields added to it already (ctx, ballot id and layer)

Reply:
wait... I just checked - so proposal, ballot and layer all have been added to the logger already. We use it only to print "new ballot" inlining and thereby logging the same information twice again - then pass it allong to `checkBallotSyntacticValidity` to then only print once `ballot is syntactically valid`

How about:
- we do not pass the logger to `checkBallotSyntacticValidity` at all and print the "syntactically valid" information after it returns without an error instead
- we do not inline the ballot at the "new ballot" log line, but instead log the ballots ID and its layer
- the logger that is passed in from `HandleSyncedBallot` has no additional fields and the one that is passed in from `Handleproposal` only has the proposal ID as additional field before passing the logger along

This still provides the same information right now but gets rid of the duplicated info that we are right now creating

Wdyt?

Reply:
I refactored the code not to pass the logger.